honeybee_radiance_command/rmtxop.py

```diff
@@ -0,0 +1,431 @@
+"""rmtxop command."""
+from .options.rmtxop import RmtxopOptions
+from ._command import Command
+import honeybee_radiance_command._exception as exceptions
+import honeybee_radiance_command._typing as typing
+
+
+class Rmtxop(Command):
+    """Rmtxop command.
+
+    Concatenate, add, multiply, divide, transpose, scale, and convert matrices. The
+    implementation here allows for matrix operations on upto three matrices at the same
+    time. While the Radiance binary allows for even more, this restriction has been
+    imposed here to keep the code somewhat clean.
+    """
+
+    __slots__ = ('_mtx1', '_mtx1_trnsps', '_mtx1_trnsfrm', '_mtx1_scl',
+                 '_mtx2', '_mtx2_trnsps', '_mtx2_trnsfrm', '_mtx2_scl',
+                 '_mtx3', '_mtx3_trnsps', '_mtx3_trnsfrm', '_mtx3_scl',
+                 '_mtx12_oprtr', '_mtx23_oprtr', '_is_one_mtx_calc', '_is_two_mtx_calc',
+                 '_is_three_mtx_calc',)
+
+    def __init__(self, options=None, output=None, mtx1=None, mtx1_trnsfrm=None,
+                 mtx1_scl=None, mtx1_trnsps=None, mtx2=None, mtx2_trnsfrm=None,
+                 mtx2_scl=None, mtx2_trnsps=None, mtx3=None, mtx3_trnsfrm=None,
+                 mtx3_scl=None, mtx3_trnsps=None, mtx12_oprtr=None, mtx23_oprtr=None,
+                 is_one_mtx_calc=False, is_two_mtx_calc=False, is_three_mtx_calc=False):
```

Comment:
I know this is still a draft but it looks to me that we might want to consider list inputs for matrices and operations between them and do the check under the hood.

```suggestion
    def __init__(self, options=None, output=None, mtxs=None, trnsfrms=None):
    
```

Reply:
@mostaphaRoudsari  Hmm…I disagree. Rmtxop is very flexible in terms of syntax, and thus allows for several potential errors. From a coding perspective, it would have been far easier for me to just have inputs as a bunch of lists but I chose to allocate individual variables for every possible option. I believe this will make both prototyping workflows, and reporting runtime errors, easier. Once we start dealing with two or three matrices, keeping up with their scaling factors, multplication/addition operators will be messy and harder to trouble shoot with lists.
In this case I prefer an ugly explicit implementation, with everything specified individually, just so that we can avoid confusion later on.

(PS: I am closing this PR and opening another one with updated files. Let's (please!) continue to discussion there...!

Reply:
Hi @built-tools! Should we have a call and go through a quick `git` tutorial together. Closing these PRs and opening new ones makes it hard for me to remember the conversation. Ideally, we should continue the conversation on the same PR - make the changes here and merge it when it is ready.